### PR TITLE
EscapeAnalysis: handle SILUndef values

### PR DESCRIPTION
**Pull request by eeckstein**
fixes a crash

https://bugs.swift.org/browse/SR-10580
rdar://problem/50279857

### CODE REVIEW

**Comment by eeckstein** [comment]
@swift-ci smoke test and merge

File: lib/SILOptimizer/Analysis/EscapeAnalysis.cpp [hunk 1]
```diff
@@ -1070,6 +1070,11 @@ static bool mayContainReference(SILType Ty, const SILFunction &F) {
 
 bool EscapeAnalysis::isPointer(ValueBase *V) {
   auto *F = V->getFunction();
+
+  // The function can be null, e.g. if V is an undef.
+  if (!F)
```
Comment (gottesmm) [inline lib/SILOptimizer/Analysis/EscapeAnalysis.cpp]: Is it possible for this to happen for anything but undef? Maybe put in an assert?

**Comment by gottesmm** [comment]
@compnerd I think this is the fix for what you were saying on the lldb bots.

File: lib/SILOptimizer/Analysis/EscapeAnalysis.cpp (hunk 1 above)
Comment (eeckstein) [reply lib/SILOptimizer/Analysis/EscapeAnalysis.cpp]: I think currently not. This is the safe default (in case we add another SILValue kind in the future)

Comment (gottesmm) [reply lib/SILOptimizer/Analysis/EscapeAnalysis.cpp]: Yes, but we won't know to update this code when someone adds something new. They won't be able to know to make the decision. With the assert, when someone adds the new SILValue, they get a hard warning that they may need to update this code.

**Pull request merged**